pkg/operator/target_config_reconciler.go

```diff
@@ -1287,3 +1540,43 @@ func (c *TargetConfigReconciler) eventHandler() cache.ResourceEventHandler {
 		DeleteFunc: func(obj interface{}) { c.queue.Add(workQueueKey) },
 	}
 }
+
+func (c *TargetConfigReconciler) checkNamepsaceMonitoringLabel() error {
+	operatorNamespace, err := c.kubeClient.CoreV1().Namespaces().Get(c.ctx, operatorclient.OperatorNamespace, metav1.GetOptions{})
+	if err != nil {
+		klog.ErrorS(err, "error fetching operator namespace")
+		return err
+	}
+	if operatorNamespace.GetLabels()[operatorclient.OpenshiftClusterMonitoringLabelKey] != operatorclient.OpenshiftClusterMonitoringLabelValue {
+		return fmt.Errorf("namespace %v is not labeled with %v=%v", operatorclient.OperatorNamespace, operatorclient.OpenshiftClusterMonitoringLabelKey, operatorclient.OpenshiftClusterMonitoringLabelValue)
+	}
+	return nil
+}
+
+func (c *TargetConfigReconciler) isSoftTainterNeeded(descheduler *deschedulerv1.KubeDescheduler) (bool, error) {
+	if descheduler.Spec.ProfileCustomizations != nil && descheduler.Spec.ProfileCustomizations.DevEnableSoftTainter {
+		return true, nil
+	}
+	nodes, err := c.kubeClient.CoreV1().Nodes().List(c.ctx, metav1.ListOptions{})
```

Comment:
This needs to be changed to read the nodes from the informer, not directly accessing the apiserver to minimize the traffic.

Reply:
done